processor/spanmetricsprocessor/processor.go

```diff
@@ -231,21 +245,40 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces pdata.Traces) e
 // writes the raw metrics data into the metrics object.
 func (p *processorImp) buildMetrics() *pdata.Metrics {
 	m := pdata.NewMetrics()
-	ilm := m.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
-	ilm.InstrumentationLibrary().SetName("spanmetricsprocessor")
+	rms := m.ResourceMetrics()
+	for key := range p.resourceAttrList {
+		p.lock.RLock()
+		resourceAttrKey := resourceKey(key)
+		resourceAttributesMap := p.resourceKeyToDimensions[resourceAttrKey]
+
+		// if the service name doesn't exist, we treat it as invalid and do not generate a trace
```

Comment:
Would this be a breaking change to those using this processor without `service.name`?

Reply:
roughly answered here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6717#discussion_r768113596

but `service.name` is defined as a mandatory resource attribute by the [OTel semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md) so it should be there. Happy to remove this if we want, it shouldn't break any logic. The main breaking change is that we are moving the service.name field from being extracted from a resource attribute and transformed into a metric normal attribute (dimension) to being extracted from a resource attribute and transformed into a metric resource attribute as default behaviour.